### PR TITLE
[15.0][IMP] scrap_reason_code: define product categories allowed for each code

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Also redefine reason code form view a bit.

Forward port of https://github.com/OCA/stock-logistics-warehouse/pull/1713.

### CODE REVIEW

File: scrap_reason_code/models/stock_scrap.py [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+        states={"done": [("readonly", True)]},
+        domain="[('id', 'in', allowed_reason_code_ids)]",
+    )
+    allowed_reason_code_ids = fields.Many2many(
+        comodel_name="scrap.reason.code",
+        compute="_compute_allowed_reason_code_ids",
     )
     scrap_location_id = fields.Many2one(readonly=True)
 
+    @api.depends("product_id")
```
Comment (JordiMForgeFlow) [inline scrap_reason_code/models/stock_scrap.py]:
```suggestion
    @api.depends("product_id.categ_id")
```

**Review by mariadforgeflow — APPROVED** [review]
LGTM!

File: scrap_reason_code/models/stock_scrap.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply scrap_reason_code/models/stock_scrap.py]: I think both are actually needed, updated!

**Review by JordiMForgeFlow — APPROVED** [review]
technical review 👍🏼

**Review by JoanSForgeFlow — APPROVED** [review]
Technical review! LGTM! :+1:

**Pull request merged**